hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -537,7 +558,9 @@ private void addPendingEvent(Event event) {
         public void run() {
             int numTimeouts = checkPendingTimeouts();
 
-            if (!pendingData.isEmpty() || !pendingEvents.isEmpty() || numTimeouts > 0) {
+            checkMissingStates();
```

Comment:
I think you are changing the design here.
I don't think 
int numMissingTooLong = checkMissingStates();

make sense, as missingState is a timer, so it's need updated and evaluated. a !missingStates.isEmpty() sounds enough to me to run the rules engine with no data, missing states needs to be in sync with MissingConditions here.

I don't see what use case are we missing in the present design (if any).